EncoreTickets.SDK/Api/Utilities/EnvironmentHelper.cs

```diff
@@ -13,20 +14,22 @@ public static class EnvironmentHelper
         /// <exception cref="ArgumentException">Thrown when the name parameter has value that is not supported.</exception>
         public static Environments EnvironmentFromName(string name)
         {
-            switch (name.ToLower())
+            try
             {
-                case "dev":
-                    return Environments.Sandbox;
-                case "qa":
-                    return Environments.QA;
-                case "staging":
-                    return Environments.Staging;
-                case "production":
-                case "prod":
-                    return Environments.Production;
-                default:
-                    throw new ArgumentException($"Environment {name} is not defined.");
-            } 
+                return EnumExtension.GetEnumFromString<Environments>(name);
+            }
+            catch (Exception)
+            {
+                switch (name.ToLower())
```

Comment:
Why is this change needed? Seems much less readable and more error-prone than it was before

Reply:
This was done to expand the available string values ​​with values ​​that correspond to the enum values ​​by their names. For example, in the previous version it was impossible to get enum for the string “sandbox”

Reply:
Is "sandbox" string even used anywhere outside the SDK? Previously we had one switch-case statement in this method, now we have try-catch blocks with the catch block containing part of the old switch-case. It has become less readable

Reply:
OK, I can get it back. I agree that this is less readable, but it is more flexible and will continue to work if new environments are added

Reply:
It would break if an enum entry gets renamed though unlike the previous solution so I would argue that it's even not really more flexible